src/ert/scheduler/scheduler.py

```diff
@@ -76,6 +78,10 @@ def __init__(
         self._ee_token = ee_token
 
     def kill_all_jobs(self) -> None:
+        assert self._loop and self._loop.is_running()
```

Comment:
why do we assert it is running? I guess it will only change which exception we get, it will crash nonetheless?

Reply:
Yes, it needs to be running otherwise we get exception - regardless. But for the sake of understanding I would keep it there.

Reply:
I suppose `assert self._loop` is there to satisfy `mypy` and for no other reason. 

Then I question if an assertionerror is more readable than an exception that says that the loop is not running. If my memory serves me right, the exception when that happens is readable.

Reply:
> I suppose `assert self._loop` is there to satisfy `mypy` and for no other reason.

yes, correct, but also not to cancel before we are running, `scheduler.execute()`.

> Then I question if an assertionerror is more readable than an exception that says that the loop is not running. If my memory serves me right, the exception when that happens is readable.

Can remove it then :+1: 
wrt. readability I meant that somebody reading the code would understand the cancellation (from another thread) can happen only if the loop is running.